### PR TITLE
appflowy: add support for darwin, modernize package and format using nixfmt

### PR DESCRIPTION
**Pull request by afh**
## Description of changes

* ℹ️ This PR builds upon ~~#325588~~#327668
* Modernize package by:
  - using `finalAttrs` rather than `rec`
  - Creating multiple directories with `mkdir`
* Format package using [NixOS/nixfmt](https://github.com/NixOS/nixfmt)
* ~~❓ I dislike that `url` and `hash` are duplicated for the darwin platform in `dist`. Can someone think of a more elegant way to do this?~~ Handled using `rec` (for details see [this comment below](https://github.com/NixOS/nixpkgs/pull/327491#discussion_r1678836047))

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/office/appflowy/default.nix [hunk 1]
```diff
@@ -1,31 +1,49 @@
...
+    {
+      x86_64-linux = {
+        urlSuffix = "linux-x86_64.tar.gz";
+        hash = "sha256-u8Y676sBLDI57z7Zpe72ELaibe83EFwB0oq+ondRIcU=";
+      };
+      x86_64-darwin = {
+        urlSuffix = "macos-universal.zip";
+        hash = "sha256-muTmxNm3dFDeBzYyZZD1GiaHYuy427tqhxz2bL3Umis=";
+      };
+      aarch64-darwin = {
```
Comment (jopejoe1) [inline pkgs/applications/office/appflowy/default.nix:26]: > ❓ I dislike that url and hash are duplicated for the darwin platform in dist. Can someone think of a more elegant way to do this?

You could set `aarch64-darwin` to `dist.x86_64-darwin`

```suggestion
      aarch64-darwin = dist.x86_64-darwin;
```

Comment (afh) [reply pkgs/applications/office/appflowy/default.nix:26]: Thanks for the suggestion, @jopejoe1, yet that results in `error: infinite recursion encountered` for me. What am I doing wrong?

Comment (jopejoe1) [reply pkgs/applications/office/appflowy/default.nix:26]: my bad, did not think about that.

changing dist to a recursive attrset (`rec`) should make it possible to set `aarch64-darwin = x86_64-darwin;`

Comment (afh) [reply pkgs/applications/office/appflowy/default.nix]: Excellent! Thanks, @jopejoe1, I've made and pushed the appropriate changes to this PR.

**Review by DarkOnion0 — APPROVED** [review]
Thank you for porting appflowy to darwin. I always wanted to do so, but I don't own a mac, so I was always a bit reluctant 😬. 

Regarding the linux parts, it works and builds as excepted. 

However, even though I like what you have done, would you mind also adding you as a maintainer? Not to do the casual update stuff, but rather if something suddenly goes wrong regarding the darwin platforms and that it needs some extended debugging.

Overall, it looks good to me :+1:, just keep in mind that I've updated appflowy to the latest version few minutes ago (#327668)!

---

Result of `nixpkgs-review pr 327491` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>appflowy</li>
  </ul>
</details>

**Comment by afh** [comment]
Thank you, @DarkOnion0, for reviewing and approving the changes in this PR, very much appreciated 😃 

At this point in time I'd like to respectfully decline the invitation to add myself as a (darwin) maintainer for appflowy and offer the following options if future issues with appflowy and the darwin platform arise:
* Feel free to involve me directly on issues or PRs
* Consider pinging [darwin-maintainers](https://github.com/orgs/NixOS/teams/darwin-maintainers) (which I'm part of)
* Think about requesting access to a [nix-community builder for darwin](https://nix-community.org/community-builder/) if you take interest in Nix on the darwin platform.

**Review by DarkOnion0 — APPROVED** [review]
Builds and runs fine, LGTM!

Result of `nixpkgs-review pr 327491` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>appflowy</li>
  </ul>
</details>

**Comment by DarkOnion0** [comment]
> At this point in time I'd like to respectfully decline the invitation to add myself as a (darwin) maintainer for appflowy and offer the following options if future issues with appflowy and the darwin platform arise:

No problem, already very much appreciate you proposed your help (and the other options) in case of something went wrong 🙂.

**Review by DarkOnion0 — APPROVED** [review]
Still works just fine

---

Result of `nixpkgs-review pr 327491` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>appflowy</li>
  </ul>
</details>

**Comment by afh** [comment]
Thanks for the helpful review comments, approval, and merging, @DarkOnion0 and @jopejoe1, much appreciated 😃

**Pull request merged**